src/test/java/org/sonatype/nexus/repository/composer/internal/ComposerFormatAttributesExtractorTest.java

```diff
@@ -0,0 +1,269 @@
+  };
+
+  @Mock
+  private Blob blob;
+
+  @Mock
+  private TempBlob tempBlob;
+
+  @Test
+  public void extractInfoFromZipballWithJson() throws Exception {
+    try (InputStream in = getClass().getResourceAsStream("extractInfoFromZipballWithJson.zip")) {
+      when(tempBlob.getBlob()).thenReturn(blob);
+      when(blob.getInputStream()).thenReturn(in);
+
+      NestedAttributesMap attributesMap = new NestedAttributesMap("composer", new LinkedHashMap<>());
+      ComposerFormatAttributesExtractor underTest = new ComposerFormatAttributesExtractor();
+      underTest.extractFromZip(tempBlob, attributesMap);
+
+      assertThat(attributesMap.keys(), containsInAnyOrder(EXPECTED_FIELDS));
+      assertThat(attributesMap.get(P_NAME), is("vendor/project"));
+      assertThat(attributesMap.get(P_VERSION), is("1.2.3"));
+      assertThat(attributesMap.get(P_DESCRIPTION), is("Test description"));
+      assertThat(attributesMap.get(P_TYPE), is("library"));
+      assertThat((List<String>) attributesMap.get(P_KEYWORDS), containsInAnyOrder("keyword1", "keyword2", "keyword3"));
+      assertThat(attributesMap.get(P_HOMEPAGE), is("http://www.example.com/"));
+      assertThat(attributesMap.get(P_TIME), is("2008-05-15"));
+      assertThat(attributesMap.get(P_LICENSE), is("MIT"));
+      assertThat((List<String>) attributesMap.get(P_AUTHORS),
+          containsInAnyOrder(
+              "Author1 <author1@example.com> (http://www.example.com/author1)",
+              "Author2 <author2@example.com> (http://www.example.com/author2)"));
+      assertThat(attributesMap.get(P_SUPPORT_EMAIL), is("email@example.com"));
+      assertThat(attributesMap.get(P_SUPPORT_ISSUES), is("irc://irc.example.com/irc"));
+      assertThat(attributesMap.get(P_SUPPORT_FORUM), is("http://www.example.com/forum"));
+      assertThat(attributesMap.get(P_SUPPORT_WIKI), is("http://www.example.com/wiki"));
+      assertThat(attributesMap.get(P_SUPPORT_SOURCE), is("http://www.example.com/source"));
+      assertThat(attributesMap.get(P_SUPPORT_DOCS), is("http://www.example.com/docs"));
+      assertThat(attributesMap.get(P_SUPPORT_RSS), is("http://www.example.com/rss"));
+    }
+  }
+
+  @Test
+  public void extractInfoFromZipballWithoutJson() throws Exception {
+    try (InputStream in = getClass().getResourceAsStream("extractInfoFromZipballWithoutJson.zip")) {
+      when(tempBlob.getBlob()).thenReturn(blob);
+      when(blob.getInputStream()).thenReturn(in);
+
+      NestedAttributesMap attributesMap = new NestedAttributesMap("composer", new LinkedHashMap<>());
+      ComposerFormatAttributesExtractor underTest = new ComposerFormatAttributesExtractor();
+      underTest.extractFromZip(tempBlob, attributesMap);
+
+      assertThat(attributesMap.keys(), is(empty()));
+    }
+  }
+
+  @Test
+  public void extractStringsMissing() {
+    NestedAttributesMap destination = new NestedAttributesMap("backing", new LinkedHashMap<>());
```

Comment:
Trivial: You could consider moving this to setup as it is used an a majority or cases (if you decide to use setup from the comment above)